hazelcast/src/main/java/com/hazelcast/util/IterableUtil.java

```diff
@@ -71,4 +72,46 @@ public void remove() {
     public static <T> Iterable<T> nullToEmpty(Iterable<T> iterable) {
         return iterable == null ? Collections.<T>emptyList() : iterable;
     }
+
+    public static <T> Iterable<T> createIterableWithSingleItem(final T item) {
```

Comment:
Is this not a bit wasteful; creating 2 objects to iterate over a single object? The problem with these generic functions is that have a specific purpose in the beginning and later on they are being re-used and then it can quickly lead to suboptimal performance.